test/ClangImporter/availability_returns_twice.swift

```diff
@@ -0,0 +1,15 @@
+// RUN: %target-swift-frontend -typecheck -verify %s
+
+#if os(OSX) || os(iOS) || os(watchOS) || os(tvOS)
+  import Darwin
+  typealias JumpBuffer = Int32
+#else
+  import Glibc
+  typealias JumpBuffer = __jmp_buf_tag
+#endif
+
+func test_unavailable_returns_twice_function() {
+  var x: JumpBuffer
```

Comment:
The particular improvement we could make is to allow `&` applied to a homogeneous tuple (today's fixed-sized array) implicitly convert to a pointer to the first element, without having to go through type rebinding. cc @atrick